docs/guides/implementation-guides/how-to-implement-dependency-injection-with-pure-di.md

```diff
@@ -0,0 +1,207 @@
+---
+id: how-to-implement-dependency-injection-with-pure-di
+title: How To Implement Dependency Injection with Pure.DI code generator
+---
+
+# How To Implement Dependency Injection in Avalonia with Pure.DI code generator
+
+This example is based on [another example](how-to-implement-dependency-injection.md) on dependency injection showing the benefits of the pure Dependency Injection approach using the Pure.DI source code generator. 
+
+## Step 0: Presentation model
+
+Compared to [this example](how-to-implement-dependency-injection.md), only a few properties and methods have been added to demonstrate real-world usage scenarios:
+
+```csharp
+public interface IMainViewModel
+{
+    string Title { get; }
+    
+    string Greetings { get; }
+}
+
+public class MainViewModel(IBusinessService businessService)
+    : IMainViewModel
+{
+    public string Title => "Avalonia application";
+
+    public string Greetings => businessService.CreateGreetings();
+}
+
+public interface IBusinessService
+{
+    string CreateGreetings();
+}
+
+public class BusinessService(IRepository repository)
+    : IBusinessService
+{
+    public string CreateGreetings()
+    {
+        repository.RegisterSomething();
+        return "Example of Dependency Injection implementation using Pure.DI";
+    }
+}
+
+public interface IRepository
+{
+    void RegisterSomething();
+}
+
+public class Repository: IRepository
+{
+    public void RegisterSomething()
+    {
+    }
+}
+```
+
+## Step 1: Install the NuGet package for DI
+
+Run the following command in a terminal inside your project directory to install the [Pure.DI](https://www.nuget.org/packages/Pure.DI) source generator package.
+
+```shell
+dotnet add package Pure.DI
+```
+
+## Step 2: Add a composition class
+
+This class setups how the composition of objects will be created for the application.
+
+```csharp
+using Pure.DI;
+using static Pure.DI.Lifetime;
+
+public partial class Composition
+{
+    private static void Setup() => DI.Setup()
+        .Root<MainWindow>(nameof(MainWindow))
+        .Root<IMainViewModel>(nameof(MainViewModel))
+
+        .Bind().As(Singleton).To<Repository>()
+        .Bind().To<BusinessService>()
+        .Bind().As(Singleton).To<MainViewModel>();
+}
+```
+
+Advantages over classical DI container libraries:
+- No performance impact or side effects when creating composition of objects.
+- All logic for analyzing the graph of objects, constructors and methods takes place at compile time. Pure.DI notifies the developer at compile time of missing or cyclic dependencies, cases when some dependencies are not suitable for injection, etc.
```

Comment:
This paragraph is just an elaborated version of the previous one

Reply:
> Advantages over classical DI container libraries:
> - No performance impact or side effects when creating composition of objects.
> - All logic for analyzing the graph of objects, constructors and methods takes place at compile time. Pure.DI notifies the developer at compile time of missing or cyclic dependencies, cases when some dependencies are not suitable for injection, etc.
> - Does not add dependencies to any additional assembly.
> - Since the generated code uses primitive language constructs to create object compositions and does not use any libraries, you can easily debug the object composition code as regular code in your application.

This is a paragraph about the general benefits.

> Advantages over classical DI container libraries:
> - No explicit initialisation of data contexts is required. Data contexts are configured directly in `.axaml` files according to the MVVM approach.
> - The code looks simple, compact and doesn't require much maintenance effort.
> - The main window is created in a pure DI paradigm, and it can be easily supplied with all necessary dependencies via DI as regular types.

This paragraph describes the advantages of using Pure.DI in Avalonia applications.

I have not combined them as they refer to different parts of the document.